### DISCUSSION TITLE
Too few arguments to function \SchemalessAttributes::__construct(), 0 passed

### DISCUSSION THREAD

**Discussion by Khaldi-Anouar** [spatie/laravel-schemaless-attributes#81] [Q&A]
Too few arguments to function Spatie\SchemalessAttributes\SchemalessAttributes::__construct(), 0 passed in ...\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Concerns\HasAttributes.php on line 1266 and exactly 2 expected

**Comment by Kessir** [comment]
I'm having the same issue on Laravel 8,  PHP8. 
For now, I have reverted to the old way of setting up:
`  public $casts = [
        'extra' => 'array'
    ];`

**Comment by Crowly34** [comment]
@Kessir @Khaldi-Anouar Issue for me here was that I was doing `use Spatie\SchemalessAttributes\SchemalessAttributes` in the casting where I should be using `use Spatie\SchemalessAttributes\Casts\SchemalessAttributes;` like the docs say hope this works for you.
